### PR TITLE
some updates and more functionality

### PR DESCRIPTION
**Pull request by THE-ORONCO**
No idea why the whole .gitignore file is flagged as changed, but I just added IDE-stuff so that IntelliJ stuff would be ignored.
- used the new match case instead of the elif stuff
- added some consisted formatting to archive.py
- added ctftime.py with the functionality to inspect CTFTime events and also register them as events in discord

### CODE REVIEW

**Comment by vincentscode** [comment]
I think the full-file differences may be due to a difference in line endings (on linux and windows). It would be great if you could fix that as it makes it easier to see what has really been changed.

**Comment by vincentscode** [comment]
Perfect, way easier to read now!
I am torn on switching to the new match-case statement. It definitely is easier to read but the bot is currently running on Python 3.9 (with match-case being introduced in 3.10). I will check how much work it would be to switch to the newer version.

File: bot.py [hunk 1]
```diff
@@ -19,38 +16,35 @@
 
 # define bot
 intents = discord.Intents().all()
-intents.message_content = True
```
Comment (vincentscode) [inline bot.py]: Removing this will not work iirc.

File: bot.py [hunk 2]
```diff
@@ -19,38 +16,35 @@
...
-
-    elif isinstance(error, commands.NoPrivateMessage):
-        await ctx.author.send(f"{ctx.command} can not be used in Private Messages.")
-
-    elif isinstance(error, commands.MissingRequiredArgument):
-        await ctx.send(f"Missing argument: {error.param}")
-
-    elif isinstance(error, commands.ChannelNotFound):
-        await ctx.send(f"Channel or Category not found: {error.argument}")
+    match error:
```
Comment (vincentscode) [inline bot.py]: Please revert back to using if / elif. Upgrading to match/case is currently not easily possible.

**Review by vincentscode — CHANGES_REQUESTED** [review]

**Comment by THE-ORONCO** [comment]
I changed bot.py back.
hopefully this fixes the problems.

**Comment by vincentscode** [comment]
unfortunately it doesn't look like you reverted all the changes

**Comment by vincentscode** [comment]
Well that seems like a bad idea. 👀

**Review by vincentscode — APPROVED** [review]

**Pull request merged**